File: gtts_service.py
from gtts import gTTS
import sys
import os

def synthesize(text, filename="summary.mp3", lang="en"):
    tts = gTTS(text=text, lang=lang)
    output_path = os.path.join("storage", "app", "public", filename)
    os.makedirs(os.path.dirname(output_path), exist_ok=True)
    tts.save(output_path)
    return output_path

if __name__ == "__main__":
    text = sys.argv[1]
    filename = sys.argv[2] if len(sys.argv) > 2 else "summary.mp3"
    path = synthesize(text, filename)
    print(path)

File: app/Services/GttsService.php
<?php

namespace App\Services;

class GttsService
{
    public function synthesize(string $text, string $filename = 'summary.mp3'): string
    {
        $pythonScript = base_path('gtts_service.py');
        $output = shell_exec("python3 {$pythonScript} " . escapeshellarg($text) . " " . escapeshellarg($filename));

        $path = trim($output);
        return asset("storage/{$filename}");
    }
}
